src/index.js

```diff
@@ -1,38 +1,108 @@
+        ${task.description}
+        <i class="fa-solid fa-ellipsis-vertical"></i>
       </p>
     `;
 
+    const optionIcon = li.querySelector('.fa-ellipsis-vertical');
+    optionIcon.addEventListener('click', (event) => {
+      event.stopPropagation(); // Prevent the event from propagating to the li element
+      optionIcon.classList.replace('fa-ellipsis-vertical', 'fa-trash-can');
+      li.style.backgroundColor = '#FFF9C4';
+      const trashIcon = li.querySelector('.fa-trash-can');
+      trashIcon.addEventListener('click', () => {
+        ul.removeChild(li);
+        tasks = tasks.filter((item) => item.index !== task.index);
+        saveTasksToLocalStorage();
+      });
+    });
+
+    const checkbox = li.querySelector('.checkbox');
+    checkbox.addEventListener('click', (event) => {
+      event.stopPropagation(); // Prevent the event from propagating to the li element
+      li.classList.toggle('overline');
+      task.completed = checkbox.checked;
+      saveTasksToLocalStorage();
+    });
+    /* eslint-disable no-use-before-define */
+    li.addEventListener('click', () => {
+      editPost(li, task);
+    });
+
     ul.appendChild(li);
   });
 
   todoList.appendChild(ul);
+}
 
-  const optionsIcons = document.querySelectorAll('.fa-ellipsis-vertical');
-  optionsIcons.forEach((icon) => {
-    icon.addEventListener('click', () => {
-      icon.classList.replace('fa-ellipsis-vertical', 'fa-trash-can');
-      const li = icon.parentElement.parentElement;
-      li.style.backgroundColor = '#FFF9C4';
-    });
+function editPost(li, task) {
+  const p = li.querySelector('p');
+  const currentDescription = task.description;
+
+  // Create an input field with the current description
+  const inputField = document.createElement('input');
+  inputField.type = 'text';
+  inputField.value = currentDescription;
+
+  // Replace the paragraph with the input field
+  p.replaceWith(inputField);
+  inputField.focus();
+
+  // Save the edited task when 'Enter' key is pressed
+  inputField.addEventListener('keypress', (event) => {
+    if (event.key === 'Enter') {
+      task.description = inputField.value.trim();
+      saveTasksToLocalStorage();
+      renderTasks();
+    }
   });
 }
 
+function addNewTask(description) {
+  const newTaskDescription = description.trim();
+  if (newTaskDescription !== '') {
+    const newTask = { description: newTaskDescription, completed: false, index: tasks.length + 1 };
+    tasks.push(newTask);
+    saveTasksToLocalStorage();
+    renderTasks();
+  }
+}
+
+const input = document.querySelector('.input-list');
+const form = document.querySelector('form');
+form.addEventListener('submit', (event) => {
+  event.preventDefault(); // Prevent form submission
+  addNewTask(input.value);
+  input.value = '';
+});
+
+const xbtn = document.getElementById('xbtn');
+xbtn.addEventListener('click', () => {
+  tasks = tasks.filter((task) => !task.completed);
+  saveTasksToLocalStorage();
+  renderTasks();
```

Comment:
- [ ] Great job working on the functionality for adding and removing but I think it would be great if you fix the indexes since when I try to remove the second task the indexes are not updated to be 1, 2, 3 it becomes 1, 3, 4 but it should be ordered as mentioned in the requirements so please fix that ✨🤩

![image](https://github.com/Nessrine88/Todolist/assets/58553711/dade07c2-f804-4630-b69a-9ecc5783fa1b)


##The indexes
![image](https://github.com/Nessrine88/Todolist/assets/58553711/600c6212-2e6f-4ed8-96f8-62fadad6e45a)

## The indexes after removing the second 


![image](https://github.com/Nessrine88/Todolist/assets/58553711/53f152be-116d-4fa2-a8da-7124dabcd662)